src/main/java/de/tum/in/www1/artemis/service/ProgrammingAssessmentService.java

```diff
@@ -61,7 +68,50 @@ public Result submitManualAssessment(long resultId) {
         Result result = resultRepository.findWithEagerSubmissionAndFeedbackAndAssessorById(resultId)
                 .orElseThrow(() -> new EntityNotFoundException("No result for the given resultId could be found"));
 
-        Double calculatedScore = calculateTotalScore(result.getFeedbacks());
+        Double calculatedScore = calculateTotalScore(result);
         return submitResult(result, result.getParticipation().getExercise(), calculatedScore);
     }
+
+    /**
+     * Calculates the total score for programming exercises.
+     * @param result with information about feedback and exercise
+     * @return calculated totalScore
+     */
+    private Double calculateTotalScore(Result result) {
+        double totalScore = 0.0;
+        double scoreAutomaticTests = 0.0;
+        ProgrammingExercise programmingExercise = (ProgrammingExercise) result.getParticipation().getExercise();
+        List<Feedback> assessments = result.getFeedbacks();
+        var gradingInstructions = new HashMap<Long, Integer>(); // { instructionId: noOfEncounters }
+
+        for (Feedback feedback : assessments) {
+            if (feedback.getGradingInstruction() != null) {
+                totalScore = gradingCriterionService.computeTotalScore(feedback, totalScore, gradingInstructions);
+            }
+            else {
+                /*
+                 * In case no structured grading instruction was applied on the assessment model we just sum the feedback credit. We differentiate between automatic test and
+                 * automatic SCA feedback (automatic test feedback has to be capped)
+                 */
+                if (feedback.getType() == FeedbackType.AUTOMATIC && !feedback.isStaticCodeAnalysisFeedback()) {
+                    scoreAutomaticTests += feedback.getCredits();
+                }
+                else {
+                    totalScore += feedback.getCredits();
+                }
+            }
+        }
+        /** Calculated score from automatic test feedbacks, is capped to max points + bonus points,
+        * see also see {@link ProgrammingExerciseGradingService#updateScore} */
+        double maxPoints = programmingExercise.getMaxScore() + Optional.ofNullable(programmingExercise.getBonusPoints()).orElse(0.0);
+        if (scoreAutomaticTests > maxPoints) {
+            scoreAutomaticTests = maxPoints;
+        }
+        totalScore += scoreAutomaticTests;
+        // Make sure to not give negative points
+        if (totalScore < 0) {
+            totalScore = 0;
+        }
```

Comment:
You should cap this again to maxPoints, as you did in the tutor-assessment-container line 442

Reply:
This is done, but in the `submitResult` call which is done in line 72 (after this calculation). Then in the `submitResult` call we call a different `calculateTotalScore` method see `AssessementService.java:227` and there we cap it to the maxScore.

Reply:
Ok